src/cost_functions.jl

```diff
@@ -1,3 +1,15 @@
+export DECostFunction, CostVData, L2Loss, Regularization
+
+type Regularization{L,P} <: DECostFunction
+  λ::L
+  penalty::P
+end
+
+function (f::Regularization)(p)
```

Comment:
I don't think we need to work around it. If that penalty is used on a scalar it's just used incorrectly.